report-ng/app/src/components/classes/classes.ts

```diff
@@ -229,20 +318,110 @@ export class Classes extends AbstractViewModel {
+                const chip: IChip = {
+                    chipType: chipType,
+                    chipElement: element
+                }
+                this._chipList.push(chip);
+            }
+        }
+
         this._filterOnce();
     }
+
+    private _removeFilter(chipType: ChipType, filterObject){
+        if(chipType == ChipType.CLEAR_ALL){
+            this._chipList = [];
+
+            delete this.queryParams.methods;
+            delete this.queryParams.q;
+            delete this.queryParams.status;
+            delete this.queryParams.class;
+            delete this.queryParams.config;
+            delete this.queryParams.failureAspect;
+            this.updateUrl(this.queryParams);
+
+            // reset class filter select options
+            this._filteredClassStatistics = [...this._executionStatistics.classStatistics.sort((a,b) => a.classIdentifier.localeCompare(b.classIdentifier))];
+
+        } else {
+
+            const index = this._chipList.map(chip => chip.chipElement).indexOf(filterObject)
+            this._chipList.splice(index, 1);
+
+            if(chipType == ChipType.CUSTOM_FILTER_TIMINGS || chipType == ChipType.CUSTOM_FILTER_FAILURE_ASPECTS){
+                // since the Custom Filter Chip only appears when navigating from another view to the tests view,
+                // we can use it for failureAspects and Methods,because they never appear together
+                delete this.queryParams.methods;
+                delete this.queryParams.failureAspect;
+            }
+
+            // insert classes that are not selected as filter anymore back into filter class select options
+            if(chipType == ChipType.CLASS && !this._filteredClassStatistics.some(stat => stat.classIdentifier == filterObject)){
+                this._filteredClassStatistics.push(this._executionStatistics.classStatistics.find(stat => stat.classIdentifier == filterObject));
+                this._filteredClassStatistics.sort((a,b) => a.classIdentifier.localeCompare(b.classIdentifier))
+            }
+
+            this.queryParams.q = this._convertQueryParams(this._chipList.filter(chip => chip.chipType === ChipType.CUSTOM_TEXT).map(chip => chip.chipElement), false);
+            this.updateUrl(this.queryParams);
+
+            if(this._chipList.length == 1){     // if there is only one chip left, it has to be removed because the last one will be the Clear All chip
+                this._chipList = [];
+            }
+        }
+        this._filter()
+    }
+
+    private _convertQueryParams(params, toArray: boolean){
+        // use "~" as separator because it is not used in id or class names and is not reserved in urls
+        if(toArray){
+            return params.split("~");
+        } else {
+            if(params == ""){
+                return;
+            }
+            return params.join("~");
+        }
+    }
+}
+
+interface IChip {
+    chipElement?: any;
```

Comment:
Since we are in the context of a chip, you can remove redundant identifiers: `chipElement` -> `element`. Also consider to set the element base type to `element: HTMLElement`